### PR TITLE
Adding function "createNew" to simplify the creation of a ezsitedata entry

### PR DESCRIPTION
**Pull request by pkamps**
Most changes are due to code reformatting. I added a new function "createNew" to simplify the creation process.
The function has a new unit test.

### CODE REVIEW

**Comment by peterkeung** [comment]
* Should be using spaces instead of hard tabs
* This should be submitted to the main eZ repo as well
* Should call the function "create" to match other persistent objects? Usually, though, only the row gets created and then you have to call "store()" outside of the class: http://share.ez.no/learn/ez-publish/a-quick-and-friendly-introduction-to-ezpersistentobject/(page)/3. That is, in fact, what the unit tests suggests (so right now you are calling "store()" twice)

**Comment by pkamps** [comment]
Ha, it's so good to have people reviewing you code.

1) yes, just yes
2) No, it's wasting my time - I don't mind if somebody else submits it.
3) I looked at other persistentobject classes like ezpolicy.php or ezrole.php: they also call 'store()' inside the class in case of the function 'createNew'. But a lot more kernel classes implement the 'create' function. So I think I'll drop 'createNew' and implement a 'create' function. Or do you think we should keep the createNew function?

**Comment by peterkeung** [comment]
2. We will discuss more internally. There is a big picture here, which is that the eZ subscription clients are using eZ's repo.
3. I like a "create" function without "store()". In keeping with the convention with normal eZ content objects, we then add "store()" in a separate API layer. It's a bit silly in my opinion, but it's at least consistent with what we use most often.

**Comment by pkamps** [comment]
Point 1) and 3) addressed.

**Comment by ernestob** [comment]
I have tested this, it works great and it is very flexible. This seems to be a good replacement for objects created in the CMS with a focus on storing data that is not going to be directly edited by the users.

Creating a value and creating a new value with the same key(name) will overwrite any existing value.

File: kernel/classes/ezsitedata.php [hunk 1]
```diff
@@ -8,38 +8,64 @@
  * @package kernel
  */
 
+/**
+ * eZPersistentObject implementation for ezsite_data table.
+ * Allows to store and fetch key/value pairs
```
Comment (peterkeung) [inline kernel/classes/ezsitedata.php]: "Allows" should be "Enables you"

File: kernel/classes/ezsitedata.php [hunk 2]
```diff
@@ -8,38 +8,64 @@
...
+            ),
+            'keys'                => array( 'name' ),
+            'class_name'          => 'eZSiteData',
+            'name'                => 'ezsite_data',
+            'function_attributes' => array()
         );
     }
 
     /**
+     * Cronstructs a new eZSiteData instance. You need to call 'store()'
```
Comment (peterkeung) [inline kernel/classes/ezsitedata.php]: "Constructs" not "Cronstructs"

**Comment by peterkeung** [comment]
Good to squash and merge.

**Comment by pkamps** [comment]
ezp pull request:
https://github.com/ezsystems/ezpublish-legacy/pull/1273

**Pull request merged**